collated/main/17navasaw.md

```diff
@@ -0,0 +1,850 @@
+    }
+}
+```
+###### \java\seedu\address\model\person\address\Unit.java
+``` java
+package seedu.address.model.person.address;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents the unit in Person's address.
+ * Guarantees: immutable; is valid as declared in {@link #isValidUnit(String)}
+ */
+public class Unit {
+    public static final String MESSAGE_UNIT_CONSTRAINTS =
+            "Unit must begin with '#' and be followed by 2 numeric strings separated by '-'";
+    private static final String UNIT_VALIDATION_REGEX = "#[\\d\\.]+-[\\d\\.]+";
+
+    public final String value;
+
+    public Unit(String unit) throws IllegalValueException {
+        requireNonNull(unit);
+        String trimmedUnit = unit.trim();
+
+        if (!isValidUnit(trimmedUnit)) {
+            throw new IllegalValueException(MESSAGE_UNIT_CONSTRAINTS);
+        }
+        this.value = trimmedUnit;
+    }
+
+    /**
+     * Returns true if a given string is a valid unit in an address.
+     */
+    public static boolean isValidUnit(String test) {
+        return test.matches(UNIT_VALIDATION_REGEX);
+    }
+}
+```
+###### \java\seedu\address\model\person\Address.java
+``` java
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
+
+    /**
+     * Validates given address.
+     *
+     * @throws IllegalValueException if given address string follows invalid format.
+     */
+    public Address(String address) throws IllegalValueException {
+        requireNonNull(address);
+        String trimmedAddress = address.trim();
+
+        if (!hasValidAddressFormat(trimmedAddress)) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+
+        splitAddressString(trimmedAddress);
+
+        this.value = trimmedAddress;
+    }
+
+    /**
+     * Splits Address into Block, Street, Unit[optional parameter] and PostalCode.
+     */
+    private void splitAddressString(String trimmedAddress) throws IllegalValueException {
+        String[] tokens = trimmedAddress.split(",");
+        int numTokens = tokens.length;
+
+        assert ((numTokens == 3) || (numTokens == 4)) : "The address should be split into 3 or 4 tokens.";
+
+        block = new Block(tokens[0]);
+        street = new Street(tokens[1]);
+
+        if (numTokens == 3) {
+
+            postalCode = new PostalCode(tokens[2]);
+
+        } else if (numTokens == 4) {
+
+            unit = new Unit (tokens[2]);
+            postalCode = new PostalCode(tokens[3]);
+        }
+    }
+
+    /**
+     * Returns true if a given string follows the valid format for address.
+     */
+    public static boolean hasValidAddressFormat(String test) {
+        StringTokenizer tokenizer = new StringTokenizer(test, ADDRESS_FORMAT_DELIMITER);
+
+        return ((tokenizer.countTokens() == 3) || (tokenizer.countTokens() == 4));
+    }
+
```

Comment:
You can make constant variables to show what these number means, and also to modularize it such that future changes are made easy.